peg-macros/analysis.rs

```diff
@@ -94,13 +94,17 @@ impl<'a> LeftRecursionVisitor<'a> {
                     .iter()
                     .position(|caller_name| caller_name == &name)
                 {
-                    let mut recursive_loop = self.stack[loop_start..].to_vec();
-                    recursive_loop.push(name);
-                    self.errors.push(LeftRecursionError {
-                        path: recursive_loop,
-                        span: rule_ident.span(),
-                    });
-                    return false;
+                    if let Some(rule) = self.rules.get(&name) {
```

Comment:
This whole block (starting with `if let Some(loop_start) = self...`) could be moved into the `if let Some(rule) = self.rules.get(&name) {` below to get access to `rule` instead of duplicating that lookup here.

Reply:
Yep, missed that, thanks!